src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        int index = random.nextInt(5);
+        switch (index) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor(),
+                        random.nextInt(10) + 1);
```

Comment:
```suggestion
            case 0:
                return getRandomSquare();
```
lets add separate methods to get each figure, this would make your code easier to read